commit 2a61169209c72317d4933f8d22f749a6a61a3d36
Author: Johan Hedberg <johan.hedberg@nokia.com>
Date:   Sat Feb 19 12:06:00 2011 -0300

    Bluetooth: Add mgmt_auth_failed event
    
    To properly track bonding completion an event to indicate authentication
    failure is needed. This event will be sent whenever an authentication
    complete HCI event with a non-zero status comes. It will also be sent
    when we're acting in acceptor role for SSP authentication in which case
    the controller will send a Simple Pairing Complete event.
    
    Signed-off-by: Johan Hedberg <johan.hedberg@nokia.com>
    Signed-off-by: Gustavo F. Padovan <padovan@profusion.mobi>

diff --git a/include/net/bluetooth/mgmt.h b/include/net/bluetooth/mgmt.h
index 5aee200e5e36..1e63c3141a78 100644
--- a/include/net/bluetooth/mgmt.h
+++ b/include/net/bluetooth/mgmt.h
@@ -264,3 +264,10 @@ struct mgmt_ev_user_confirm_request {
 	bdaddr_t bdaddr;
 	__le32 value;
 } __packed;
+
+#define MGMT_EV_AUTH_FAILED		0x0010
+struct mgmt_ev_auth_failed {
+	__le16 index;
+	bdaddr_t bdaddr;
+	__u8 status;
+} __packed;